engine/jobmaster/dm/worker_manager.go

```diff
@@ -280,6 +283,29 @@ func (wm *WorkerManager) checkAndScheduleWorkers(ctx context.Context, job *metad
 			recordError = err
 			continue
 		}
+
+		// create success, record unit state
+		if err = wm.unitStore.ReadModifyWrite(ctx, func(state *metadata.UnitState) error {
+			status, ok := state.CurrentUnitStatus[taskID]
+			if !ok {
+				state.CurrentUnitStatus[taskID] = &metadata.TaskStatus{
+					Unit:        nextUnit,
+					Task:        taskID,
+					CreatedTime: time.Now(),
+					// we don't care about the Stage and CfgModRevision here
```

Comment:
If so, Please define a new struct for it. e.g. Struct{Unit; CreatedTime;}

Reply:
In engine, task is a definition  so similar to unit. Do you also think we should new a struct to manage it? @lance6716